event/src/test/java/io/airlift/event/client/TestLoggingEventClient.java

```diff
@@ -0,0 +1,146 @@
+        EventConfig config = new EventConfig();
+        MockRollingFileAppender appender = new MockRollingFileAppender();
+        LoggingEventClient client = createLoggingEventClient(config, appender);
+        client.postEvent("test event");
+        assertEquals(appender.getLogs().size(), 0, "client shouldn't log any events");
+    }
+
+    @Test
+    public void testEventLogging() throws IOException
+    {
+        EventConfig config = new EventConfig();
+        config.setLogEnabled(true);
+        config.setLogLayoutClass(MockLayout.class.getName());
+        config.setLogPath(temp.getAbsolutePath());
+        config.setEventTypeToLog("MockEvent");
+        MockRollingFileAppender appender = new MockRollingFileAppender();
+        LoggingEventClient client = createLoggingEventClient(config, appender);
+        client.postEvent(new MockEvent());
+        assertEquals(appender.getLogs().size(), 1, "client should log a single event");
+        assertEquals(appender.getLogs().get(0).getClass(), MockEvent.class, "client should log a single event");
+    }
+
+    @Test(expectedExceptions = RuntimeException.class, expectedExceptionsMessageRegExp = "java.lang.ClassNotFoundException: nonexistent_class.*")
+    public void testNonexistentLogLayoutClass()
+    {
+        EventConfig config = new EventConfig();
+        config.setLogEnabled(true);
+        config.setLogPath(temp.getAbsolutePath());
+        config.setEventTypeToLog("MockEvent");
+        config.setLogLayoutClass("nonexistent_class");
+        MockRollingFileAppender appender = new MockRollingFileAppender();
+        createLoggingEventClient(config, appender);
+    }
+
+    @Test(expectedExceptions = NullPointerException.class, expectedExceptionsMessageRegExp = "event type to log is null")
+    public void testNullEventType()
+    {
+        EventConfig config = new EventConfig();
+        config.setLogEnabled(true);
+        config.setLogPath(temp.getAbsolutePath());
+        config.setEventTypeToLog(null);
+        config.setLogLayoutClass(MockLayout.class.getName());
+        MockRollingFileAppender appender = new MockRollingFileAppender();
+        createLoggingEventClient(config, appender);
+    }
+
+    @Test(expectedExceptions = NullPointerException.class, expectedExceptionsMessageRegExp = "event log layout class is null")
+    public void testNullLogLayout()
+    {
+        EventConfig config = new EventConfig();
+        config.setLogEnabled(true);
+        config.setLogPath(temp.getAbsolutePath());
+        config.setEventTypeToLog("MockEvent");
+        config.setLogLayoutClass(null);
+        MockRollingFileAppender appender = new MockRollingFileAppender();
+        createLoggingEventClient(config, appender);
+    }
+
+    private LoggingEventClient createLoggingEventClient(EventConfig config, MockRollingFileAppender appender)
+    {
+        LoggingEventClient client = new LoggingEventClient(config);
+        client.setFileAppender(appender);
```

Comment:
appender gets initialized in the constructor but also passed here. Does this overwrite the one from the constructor? Why multiple ways to pass this?